mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go

```diff
@@ -324,6 +365,53 @@ func generateBigtableGCPolicy(d *schema.ResourceData) (bigtable.GCPolicy, error)
 	return policies[0], nil
 }
 
+func getGCPolicyFromJSON(p map[string]interface{}) (bigtable.GCPolicy, error) {
+	policies := []bigtable.GCPolicy{}
+	if p["rules"] == nil {
+		return bigtable.NoGcPolicy(), nil
+	}
+
+	for _, raw := range p["rules"].([]interface{}) {
+		r := raw.(map[string]interface{})
+		if r["max_age"] != nil {
+			maxAge := r["max_age"].(string)
+			duration, err := time.ParseDuration(maxAge)
+			if err != nil {
+				return nil, fmt.Errorf("invalid duration string: %v", maxAge)
+			}
+			policies = append(policies, bigtable.MaxAgePolicy(duration))
+		}
+
+		if r["max_version"] != nil {
+			version := r["max_version"].(float64)
+			policies = append(policies, bigtable.MaxVersionsPolicy(int(version)))
+		}
+
+		if r["mode"] != nil {
+			n, err := getGCPolicyFromJSON(r)
+			if err != nil {
+				return nil, err
+			}
+			// Don't combine NoGcPolicy with other types of policy
+			if n != nil && n != bigtable.NoGcPolicy() {
+				policies = append(policies, n)
+			}
+		}
+	}
+
+	switch p["mode"] {
+	case strings.ToLower(GCPolicyModeUnion):
+		return bigtable.UnionPolicy(policies...), nil
+	case strings.ToLower(GCPolicyModeIntersection):
+		return bigtable.IntersectionPolicy(policies...), nil
+	default:
+		if len(policies) == 0 {
+			return bigtable.NoGcPolicy(), nil
+		}
+		return policies[0], nil
```

Comment:
For no `mode` case, the only makes sense is there is only one rule. For everything else, it should be an invalid input.